src/net_processing.cpp

```diff
@@ -1972,6 +1989,11 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                     Misbehaving(pfrom->GetId(), nDoS);
                 }
                 LogPrintf("Peer %d sent us invalid header via cmpctblock\n", pfrom->GetId());
+                // Send a getheaders message with the hash of the invalid block to all peers.
+                // Those who respond to this message will be banned.
+                connman->ForEachNode([cmpctblock, connman](CNode* pnode) {
+                    connman->PushMessage(pnode, CNetMsgMaker(pnode->GetSendVersion()).Make(NetMsgType::GETHEADERS, CBlockLocator({cmpctblock.header.GetHash()}), uint256()));
```

Comment:
Feedback loop here.

Reply:
how so?

Reply:
Node sends you bad header, you request that header from all nodes (including the one that just sent it). Same node replies with same header, you request it again from all nodes... repeat until they get banned.

Reply:
Isn't the node that sent it to you in the first place already banned by the time this runs?

Reply:
No, not necessarily. See "high-hash" for example. But even still, the ban/disconnect should not be assumed. What if they're whitelisted?